oracle.go

```diff
@@ -33,100 +31,101 @@ DROP USER {{username}};
 	defaultRotateCredsSql = `ALTER USER {{username}} IDENTIFIED BY "{{password}}"`
 )
 
+var _ dbplugin.Database = &Oracle{}
+
 type Oracle struct {
 	*connutil.SQLConnectionProducer
-	credsutil.CredentialsProducer
 }
 
-// New implements builtinplugins.BuiltinFactory
 func New() (interface{}, error) {
 	db := new()
 	// Wrap the plugin with middleware to sanitize errors
-	dbType := dbplugin.NewDatabaseErrorSanitizerMiddleware(db, db.SecretValues)
+	dbType := dbplugin.NewDatabaseErrorSanitizerMiddleware(db, db.secretValues)
 	return dbType, nil
 }
 
 func new() *Oracle {
 	connProducer := &connutil.SQLConnectionProducer{}
 	connProducer.Type = oracleTypeName
 
-	credsProducer := &oracleCredentialsProducer{
-		credsutil.SQLCredentialsProducer{
-			DisplayNameLen: oracleDisplayNameMaxLength,
-			RoleNameLen:    oracleDisplayNameMaxLength,
-			UsernameLen:    oracleUsernameLength,
-			Separator:      "_",
-		},
-	}
-
 	dbType := &Oracle{
 		SQLConnectionProducer: connProducer,
-		CredentialsProducer:   credsProducer,
 	}
 
 	return dbType
 }
 
-// Run instantiates an Oracle object, and runs the RPC server for the plugin
-func Run(apiTLSConfig *api.TLSConfig) error {
-	dbType, err := New()
+func (o *Oracle) Initialize(ctx context.Context, req dbplugin.InitializeRequest) (dbplugin.InitializeResponse, error) {
+	err := o.SQLConnectionProducer.Initialize(ctx, req.Config, req.VerifyConnection)
 	if err != nil {
-		return err
+		return dbplugin.InitializeResponse{}, err
 	}
-
-	dbplugin.Serve(dbType.(dbplugin.Database), api.VaultPluginTLSProvider(apiTLSConfig))
-
-	return nil
-}
-
-func (o *Oracle) Type() (string, error) {
-	return oracleTypeName, nil
+	resp := dbplugin.InitializeResponse{
+		Config: req.Config,
+	}
+	return resp, nil
 }
 
-func (o *Oracle) CreateUser(ctx context.Context, statements dbplugin.Statements, usernameConfig dbplugin.UsernameConfig, expiration time.Time) (username string, password string, err error) {
-	statements = dbutil.StatementCompatibilityHelper(statements)
-
-	if len(statements.Creation) == 0 {
-		return "", "", dbutil.ErrEmptyCreationStatement
+func (o *Oracle) NewUser(ctx context.Context, req dbplugin.NewUserRequest) (dbplugin.NewUserResponse, error) {
+	statements := removeEmpty(req.Statements.Commands)
+	if len(statements) == 0 {
+		return dbplugin.NewUserResponse{}, dbutil.ErrEmptyCreationStatement
 	}
 
-	// Grab the lock
 	o.Lock()
 	defer o.Unlock()
 
-	username, err = o.GenerateUsername(usernameConfig)
+	username, err := credsutil.GenerateUsername(
+		credsutil.DisplayName(req.UsernameConfig.DisplayName, 8),
+		credsutil.RoleName(req.UsernameConfig.RoleName, 8),
+		credsutil.MaxLength(30),
```

Comment:
Should these be using `oracleDisplayNameMaxLength` and `oracleUsernameLength`? Or maybe those constants should be removed.

Reply:
Removed as I prefer those values to be close to where they're used rather than separated at the top of the file.